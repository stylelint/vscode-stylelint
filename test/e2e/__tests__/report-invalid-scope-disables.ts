import { openDocument, waitForDiagnostics, assertDiagnostics, closeAllEditors } from '../helpers';

describe('"stylelint.reportInvalidScopeDisables" setting', () => {
	afterEach(async () => {
		await closeAllEditors();
	});

	it('should report invalid-scope disables when enabled', async () => {
		const { document } = await openDocument('report-disables/invalid-scope.css');
		const diagnostics = (await waitForDiagnostics(document)).filter(
			({ code }) => code === '--report-invalid-scope-disables',
		);

		assertDiagnostics(diagnostics, [
			{
				code: '--report-invalid-scope-disables',
				message: 'Rule "foo" isn\'t enabled',
				range: [0, 0, 0, 36],
				severity: 'error',
			},
			{
				code: '--report-invalid-scope-disables',
				message: 'Rule "foo" isn\'t enabled',
				range: [2, 0, 2, 31],
				severity: 'error',
			},
			{
				code: '--report-invalid-scope-disables',
				message: 'Rule "foo" isn\'t enabled',
				range: [4, 0, 4, 26],
				severity: 'error',
			},
		]);
	});
});
