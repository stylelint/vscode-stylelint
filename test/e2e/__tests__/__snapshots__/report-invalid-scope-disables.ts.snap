// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"stylelint.reportInvalidScopeDisables" setting should report invalid-scope disables when enabled 1`] = `
[
  {
    "code": "--report-invalid-scope-disables",
    "message": "Rule "foo" isn't enabled",
    "range": {
      "end": {
        "character": 36,
        "line": 0,
      },
      "start": {
        "character": 0,
        "line": 0,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
  {
    "code": "--report-invalid-scope-disables",
    "message": "Rule "foo" isn't enabled",
    "range": {
      "end": {
        "character": 31,
        "line": 2,
      },
      "start": {
        "character": 0,
        "line": 2,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
  {
    "code": "--report-invalid-scope-disables",
    "message": "Rule "foo" isn't enabled",
    "range": {
      "end": {
        "character": 26,
        "line": 4,
      },
      "start": {
        "character": 0,
        "line": 4,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
]
`;
